### PR TITLE
Fix/2514 invalid from http redirection

### LINKED ISSUE #2514: [🐛 | Bug]: Invalid `from_http_redirection` rust implementation
### Version

N/A

### What platforms you were running our software when the bug happened ?

Linux

### What happened ?

the check for `redirect` in url path is invalid

https://github.com/Scille/parsec-cloud/blob/de94b157ee8dabb0b029afb143845200b8618191/oxidation/crates/parsec_api_types/src/addr.rs#L67-L73

[`Url::path_segments`](https://docs.rs/url/latest/url/struct.Url.html#method.path_segments) would return `Some("")` when the path is empty.

So every `url` that contain something other than an empty path would pass the check

### Relevant output

consider the following test case for `BackendAddr`

```rust
#[rstest]
#[case("https://foo.bar/", false)]
#[case("https://foo.bar/redirection", false)]
#[case("https://foo.bar/redirect", true)]
#[case("https://foo.bar/not_valid", false)]
fn test_faulty_addr_redirection(#[case] raw_url: &str, #[case] is_valid: bool) {
    let res = dbg!(BackendAddr::from_http_redirection(raw_url));

    assert_eq!(res.is_ok(), is_valid)
}
```

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
closes #2514

### CODE REVIEW

File: oxidation/crates/parsec_api_types/src/addr.rs [hunk 1]
```diff
@@ -68,7 +69,7 @@ macro_rules! impl_common_stuff {
                 let mut path_segments = parsed
                     .path_segments()
                     .ok_or_else(|| "Not a redirection URL")?;
-                if path_segments.next() == Some("") && path_segments.next() == Some("redirection") {
+                if path_segments.next() != Some("redirect") {
```
Comment (touilleMan) [inline oxidation/crates/parsec_api_types/src/addr.rs:72]: Good catch !

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: oxidation/crates/parsec_api_types/tests/test_addr/test_from_http_redirect.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+use rstest::rstest;
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_types/tests/test_addr/test_from_http_redirect.rs]: done

[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_types/tests/test_addr/test_from_http_redirect.rs]: but i prefer smaller files

File: oxidation/crates/parsec_api_types/tests/test_addr/test_from_http_redirect.rs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+use parsec_api_types::BackendAddr;
+
+#[rstest]
+#[case("https://foo.bar", false)]
+#[case("https://foo.bar/redirection", false)]
+#[case("https://foo.bar/redirect", true)]
```
[the remark this answers is not in the document]
Comment (FirelightFlagboy) [reply oxidation/crates/parsec_api_types/tests/test_addr/test_from_http_redirect.rs]: done

**Review by touilleMan — APPROVED** [review]
👍

**Pull request merged**